### DISCUSSION TITLE
Address the "automatic download" concerns of LunarVim users

### DISCUSSION THREAD

**Discussion by sidkshatriya** [LunarVim/LunarVim#2836] [Q&A] [answered]
I am a prospective LunarVim user. However, I am feeling a bit hesitant to try it out. Perhaps, someone can help me with some of my concerns.

I find the core proposition of LunarVim very sound: LunarVim tries to be a batteries added distribution of NeoVim. In that sense it is like the Spacemacs distribution for emacs.

LunarVim promises to basically automatically download and install language servers. Actually I don't want my editor to be downloading binaries from the Internet as much as possible. If it has to, then it should prompt me before doing so.

**I am only interested in the good default configuration aspect of LunarVim. I am quite happy to install the language server myself (and other programs like fzf etc.)**

So, can anybody answer me:
* Is it possibly to force LunarVim to _never_ download a binary/LSP itself?
* If it is not possible to do the above does LunarVim always prompt the user before downloading something or running npm etc. ?
* Also if LunarVim is sensitive about the above topics, can the answers to these queries be added to the install documentation?  I'm sure there are a lot of users that would be mainly attracted to the good default config of LunarVim but not be so excited about the automatic install aspect...

Thanks in advance !

**Comment by rebuilt** [comment, marked as answer]
```
lvim.lsp.automatic_servers_installation = false
```

**Comment by Uzaaft** [comment]
@sidkshatriya  
Just out of curiosity, when you discovered this(the automatic server installation part), where did you look to see if this was the case? Might be useful for us to know, so that we can improve the docs.

**Reply by rebuilt** [reply]
It's probably from this section. 
 https://www.lunarvim.org/languages/#installing-and-updating-a-server

**Comment by sidkshatriya** [comment]
@Uzaaft Sorry I don't recall where I read this first. I do recall reading that automatic installation of LSP is one of LunarVim's prominent features. As mentioned above, I am more interested in fact that LunarVim comes with default built in configuration so a user does not need to tediously build it out themselves.

However, [here might be a good location](https://www.lunarvim.org/02-after-install.html#language-server) to add some text regarding only getting the LSP working but not downloading LSP binaries.

So while I have put https://github.com/LunarVim/LunarVim/discussions/2836#discussioncomment-3212196 in my `config.lua` file... I don't know how to get `clangd` working for instance on my C++ source files. Sorry if this is obvious but can anybody point out what I would need to do next?

**Reply by sidkshatriya** [reply]
I have tried doing `:LspStart, :LspRestart` etc. I suppose I need to tell LunarVim that cpp files should be associated with the clangd LSP. How do I do this easily (without installing the clangd server which I already have installed)

**Reply by rebuilt** [reply]
I'm on my phone so I can only describe the process in general terms.  You have to add clang to the list of skipped servers for autoconfiguration.  Then you configure the server yourself according to the documentation provided by the lspconfig plugin.   The specific option you want to pay attention to is the specified cmd which is how you tell lspconfig where to look for the command.   

Alternatively, you can just make sure you add the LSP folder to your path.  If you can run the LSP from the command line, lspconfig should be able to use it automatically
